### PR TITLE
Soften ValueError for TPU autodetection

### LINKED ISSUE #39913: [Core] TPU detection code can cause ray start to fail
### What happened + What you expected to happen

See https://discuss.ray.io/t/ray-init-tries-to-detect-tpus-even-when-they-arent-present/12236

Looks like it can return the empty or None value. I verified in my local machine, the request itself failed. 

### Versions / Dependencies

master

### Reproduction script

n/a

### Issue Severity

None

### PR DESCRIPTION
**Pull request by allenwang28**
## Why are these changes needed?

Softens the assertion for valid TPU type to a log in TPU autodetection.

The original TPU auto detection incorrectly assumed that a GCE call would not return 200, leading to an issue where `ray.init` may fail if it returns status code 200 and an empty string.

This should, however, fail in `gcp/config.py` because we want to assert and fail ASAP if given a malformed accelerator type. We migrate the ValueError there.

We also add a specific test case here to assert that it does not crash.

## Related issue number

Closes https://github.com/ray-project/ray/issues/39913

## Checks

- [x ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/accelerator.py
Comment (architkulkarni) [inline python/ray/_private/accelerator.py:231]: There are two ways for `detected_tpu_version` to be `None`:
1. `accelerator_type` was specified, but but it wasn't a valid accelerator type
2. `accelerator_type was not specified, but the GCE response didn't include a valid accelerator type

Do we need to distinguish these two cases in the info log?

File: python/ray/_private/accelerator.py
Comment (architkulkarni) [inline python/ray/_private/accelerator.py:209]: Just want to confirm that by introducing this `else` block, polling the GCE endpoint is no longer on the critical path of `ray.init()`?  In other words, the GCE endpoint is only polled if `RAY_GKE_TPU_ACCELERATOR_TYPE_ENV_VAR` is set by the user.

**Review by architkulkarni — COMMENTED** [review]
Overall looks good to me, just a couple things I wanted to confirm.

**Comment by architkulkarni** [comment]
Assigning @rkooo567 as ray-core codeowner

File: python/ray/_private/accelerator.py
Comment (allenwang28) [reply python/ray/_private/accelerator.py:209]: No, if the env var RAY_GKE_TPU_ACCELERATOR_TYPE_ENV_VAR is not set, then we would poll the GCE endpoint. If it is set, we don't poll the endpoint.

The difference is now if the poll fails, then nothing happens - we just log that we couldn't detect the TPU type (vs. raising an error like we did before)

Comment (architkulkarni) [reply python/ray/_private/accelerator.py:209]: Actually, I misread the code, it seems we only poll the GCE endpoint if the env var is *not* set by the user.  So in this case, `requests.get()` is still on the path of the default `ray.init()` call.  Would it make sense to gate this behind another environment variable?

Comment (architkulkarni) [reply python/ray/_private/accelerator.py:209]: Ah sorry just saw your response

Comment (architkulkarni) [reply python/ray/_private/accelerator.py:209]: Want to get your and @rkooo567's thoughts on gating the `requests.get()` call, I'm not sure how risky it is to leave it on the default path (like if it could cause hangs or cause `ray.init()` to become slow)

File: python/ray/_private/accelerator.py
Comment (architkulkarni) [reply python/ray/_private/accelerator.py:231]: (This seems relatively unimportant and doesn't need to block the PR)

Comment (allenwang28) [reply python/ray/_private/accelerator.py:250]: Added in more logging anyways!

File: python/ray/_private/accelerator.py
Comment (allenwang28) [reply python/ray/_private/accelerator.py:215]: One way we could gate the requests.get is by only running the function when we're sure we have a particular accelerator type, e.g. [here](https://github.com/ray-project/ray/blob/baa861dc5a7967dd903d1b64e88277e3e76d6047/python/ray/_private/accelerator.py#L109-L117) by replacing [this](https://github.com/ray-project/ray/blob/baa861dc5a7967dd903d1b64e88277e3e76d6047/python/ray/_private/accelerator.py#L49) with an executable function, but honestly it's not very elegant.

I don't think requests.get should have drastic impact, but then again it already has 😛 happy to hear further thoughts though.

Comment (architkulkarni) [reply python/ray/_private/accelerator.py:215]: If the logging goes straight to a file and not to the terminal output though, I think that could be acceptable.  But we wouldn't want it to be too hidden, for those people who actually need to see those logs

Comment (allenwang28) [reply python/ray/_private/accelerator.py:215]: Maybe for now we could just go with the approach for the autodetect function to trigger only if we detected an accelerator already?

Comment (architkulkarni) [reply python/ray/_private/accelerator.py:215]: That sounds good to me!

Comment (allenwang28) [reply python/ray/_private/accelerator.py:216]: Done!

**Review by architkulkarni — APPROVED** [review]
Looks good to me, nice fix!

**Comment by architkulkarni** [comment]
Failed tests:
- serve:test_multiplex 
- serve:test_proxy_state
- serve:test_max_replicas_per_node   
- serve:test_new_handle_api  
- serve:test_new_handle_api_set_via_env_var
- serve:test_persistence 
- rllib:examples/nested_action_spaces_ppo_torch
- air:test_legacy_dataset_config  

These are all unrelated. Linkcheck failure is unrelated because this PR doesn't touch the docs.

File: python/ray/_private/accelerator.py
Comment (rkooo567) [reply python/ray/_private/accelerator.py:216]: So, the requirement here is; 
1. We should avoid using request.get which is blocking
2. We should not print TPU not detected logs to users when they don't use TPUs.

Haven't followed it closely here. Is this PR satisfying these 2 requirements?

Comment (architkulkarni) [reply python/ray/_private/accelerator.py:216]: 1. Yes on the default path.  But if TPUs are detected, we do use `request.get()` to get the type of TPUs.
2. Yes

Comment (rkooo567) [reply python/ray/_private/accelerator.py:216]: That sounds perfect!

**Pull request merged**
